components/brave_shields/browser/brave_shields_util.h

```diff
@@ -19,73 +19,91 @@ struct Referrer;
 
 class GURL;
 class HostContentSettingsMap;
-class Profile;
+class PrefService;
 
 namespace brave_shields {
 
+class ReferrerWhitelistService;
+
 enum ControlType { ALLOW = 0, BLOCK, BLOCK_THIRD_PARTY, DEFAULT, INVALID };
 
 ContentSettingsPattern GetPatternFromURL(const GURL& url);
 std::string ControlTypeToString(ControlType type);
 ControlType ControlTypeFromString(const std::string& string);
 
-void SetBraveShieldsEnabled(Profile* profile, bool enable, const GURL& url);
+void SetBraveShieldsEnabled(HostContentSettingsMap* map,
+                            PrefService* local_state,
```

Comment:
maybe we can make this a last parameter with a default value? Just to make the interface less ugly